### PR TITLE
Add -fstandalone-debug debug flag

### PR DESCRIPTION
**Pull request by drisspg**
# Summary

While debugging something in lldb, I found that the formatter I wrote for c10::intarrayref was not working correctly producing:
`(std::string) $6 = error: summary string parsing error`

Based off of this thread: https://github.com/vadimcn/codelldb/issues/415

I adde the standalone-debug information and fixed the std::string formatting issue.

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -908,6 +908,9 @@ if(NOT MSVC)
   endif()
   append_cxx_flag_if_supported("-Wno-unused-but-set-variable" CMAKE_CXX_FLAGS)
   append_cxx_flag_if_supported("-Wno-maybe-uninitialized" CMAKE_CXX_FLAGS)
+  if("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+    string(APPEND CMAKE_CXX_FLAGS_DEBUG " -fstandalone-debug")
+  endif()
```
Comment (malfet) [inline CMakeLists.txt:913]:
```suggestion
  append_cxx_flag_if_supported("-fstandalone-debug" CMAKE_CXX_FLAGS_DEBUG)
```

**Review by malfet — CHANGES_REQUESTED** [review]
Just use `append_cxx_flag_if_supported`, because not all versions of clang will support it

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**